nilearn/_utils/estimator_checks.py

```diff
@@ -483,6 +490,18 @@ def check_masker_detrending(estimator):
     assert_raises(AssertionError, assert_array_equal, detrended_signal, signal)
 
 
+def check_masker_validation_mask(estimator):
+    """Raise error when mask_img is not binary."""
+    if accept_niimg_input(estimator):
+        mask_img = _img_3d_mni()
+    else:
+        mask_img = _make_surface_img()
+    estimator.mask_img = mask_img
+
+    with pytest.raises(ValueError, match="Given mask is not made of 2 values"):
+        estimator.fit()
```

Comment:
Is it obvious that any error would come from the mask not being binary ?

Reply:
this may be too strict but I believe that mask images that are passed should be 'binarisable' (as in all values should be evaluable as a truthy or falsy)

so probably wanna check non finite values (nan, inf...)

also having a check that the fitted mask (self.mask_img_) is a binary mask would make sense

Reply:
> this may be too strict but I believe that mask images that are passed should be 'binarisable' (as in all values should be evaluable as a truthy or falsy)
> 
> so probably wanna check non finite values (nan, inf...)
> 
> also having a check that the fitted mask (self.mask_img_) is a binary mask would make sense

Will do this in a follow up PR: this one had grown too big